### ISSUE TITLE
Feature Request: Add `keras.ops.connected_component` for Image Processing

### ISSUE THREAD

**Issue by innat** [keras-team/keras#23236]
**Is your feature request related to a problem?**

There is currently no built-in operation for finding connected components in binary images or special cases. This is a common operation in computer vision and graph processing. Currently, user need to rely on backend-specific APIs or third-party libraries, making it harder to write portable Keras code. Say,

- tf: Uses [tfa.image.connected_components](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/connected_components.py) (**deprecated**)
- torch: Typically relies on libraries such as [Kornia](https://github.com/kornia/kornia/blob/main/kornia/contrib/connected_components.py) or custom implementations.
- jax: Maybe requires a custom implementation.

As a result, code written with `keras.ops` often has to fall back to backend-specific branches for this operation.

**Proposed solution**

Add a new operation to `keras.ops`:

```python
keras.ops.connected_component(
    inputs,
    connectivity=1,
    background=0,
    return_num=False,
    return_stats=False,
)
```
Arguments
- inputs: Binary or integer tensor. Shapes such as (..., H, W) or (..., D, H, W).
- connectivity: Connectivity definition (4/8 for 2D, 6/18/26 for 3D).
- background: Background value.
- return_num: Whether to also return the number of connected components.
- return_stats: Whether to compute region statistics.

Returns
- Default: `labels`
- With `return_num=True`: `labels, num_components`
- With `return_stats=True`: `labels, num_components, stats`

where `stats` could be a dictionary such as

```bash
{
    "area": ...,
    "bbox": ...,
    "centroid": ...,
}
```

**Alternatives considered**

- Scipy - [label](https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.label.html)
- OpenCV - [connectedComponents](https://docs.opencv.org/4.5.1/d3/dc0/group__imgproc__shape.html#gaedef8c7340499ca391d459122e51bef5)
- https://github.com/zsef123/Connected_components_PyTorch ([YACCLAB : Yet Another Connected Components Labeling Benchmark](https://github.com/prittt/YACCLAB))

**Why this would be useful**

Connected component labeling is a fundamental operation used in many domains, including:

- Medical image analysis (counting lesions, tumors, memograph, cells, or anatomical structures in 2D/3D segmentation masks)
- Instance segmentation post-processing
- OCR pipelines (separating characters or connected text regions)
- Graph processing and GNNs (finding connected subgraphs or clusters)

**Another thing worth considering: 3D**

I would strongly recommend designing this for N-dimensional data, not just 2D. Typical shapes would become

```bash
(..., H, W)
(..., D, H, W)
```
and the connectivity parameter could naturally extend to
```
- 2D: 4, 8
- 3D: 6, 18, 26
```

However, this might take lots of effort and challenges. Current best alternative: https://github.com/seung-lab/connected-components-3d

**Alternative APIs**

Instead of 

```python
keras.ops.connected_component(
    inputs,
    connectivity=1,
    background=0,
    return_num=False,
    return_stats=False,
)
```

Might be better 

```python
labels = keras.ops.connected_component(mask)
num = keras.ops.num_connected_components(labels)
boxes = keras.ops.bounding_boxes(labels)
areas = keras.ops.component_area(labels)
centroids = keras.ops.component_centroids(labels)
```

**Comment by hertschuh** [comment]
@innat 

Thanks for the detailed write-up on this feature request.

In the absolute, I agree this would be a good addition. However, the main Keras repo may or may not be a place for this.

First, this might be a lot of code, more than what we want for a single op.

Second, I don't think this lends itself well to being implemented on top of keras ops because it's `for` loops and a lot of `if`s. Both the tf-addons and the seung-lab implementations are in C++. I fear that it will run very slow, both on CPU and on accelerator like GPU. It might just be faster to use an existing C++ implementation.

**Comment by VaishnaviMudaliar** [comment]
Hey! @innat @mrinalghoshh @dhantule , I am looking for some issues to contribute to, is this open for contribution?

**Comment by innat** [comment]
@VaishnaviMudaliar 
As @hertschuh pointed out, this is a useful image processing operation. However, it's true that implementing it faithfully is a fairly large undertaking and is likely beyond the current scope. There are already several mature alternatives available, so a native `keras.ops.connected_component` implementation would need to be competitive in both performance and functionality, which introduces significant implementation challenges.

If you have a proof-of-concept implementation or an alternative algorithm that could realistically be integrated into core Keras, it would be great if you could share it.
